app/code/Magento/Shipping/Controller/Adminhtml/Order/ShipmentProvider.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Shipping\Controller\Adminhtml\Order;
+
+use Magento\Framework\App\RequestInterface;
+
+class ShipmentProvider
```

Comment:
Shipment Provide should be extension point. 
Default Interface should be introduced to have an ability  - overwrite it in MSI

newly created interface should be marked with `@api`